src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -174,6 +193,47 @@ private static async Task HandleSingleTypeAsync(CompletionContext context, Seman
                         filterText: memberDisplayName));
                 }
             }
+            else if (enclosingNamedType is not null)
+            {
+                // Build a list of the members with the same type as the target
+                foreach (var member in type.GetMembers())
+                {
+                    ISymbol staticSymbol;
+                    ITypeSymbol symbolType;
+                    if (member is IFieldSymbol { IsStatic: true } field)
+                    {
+                        staticSymbol = field;
+                        symbolType = field.Type;
+                    }
+                    else if (member is IPropertySymbol { IsStatic: true, IsIndexer: false } property)
+                    {
+                        staticSymbol = property;
+                        symbolType = property.Type;
+                    }
+                    else
+                    {
+                        // Only fields and properties are supported for static member matching
+                        continue;
+                    }
+
+                    if (!SymbolEqualityComparer.Default.Equals(type, symbolType)
```

Comment:
Should this also be limited to read-only fields and properties (get-only)? The original intent seems to be to treat such members as (immutable) enums, though off hand I don't see any major reason to also limit this to read-write static fields or properties if used as, say, defaults. Just wanted to raise the question, though.

Reply:
Right now it doesn't restrict to read-only, but that shouldn't be a problem.